website/src/libs/api-v3/payments.js

```diff
@@ -130,33 +128,34 @@ api.createSubscription = function createSubscription (data, cb) {
 /**
  * Sets their subscription to be cancelled later
  */
-api.cancelSubscription = function cancelSubscription (data, cb) {
+api.cancelSubscription = function cancelSubscription (data) {
   let plan = data.user.purchased.plan;
   let now = moment();
   let remaining = data.nextBill ? moment(data.nextBill).diff(new Date(), 'days') : 30;
+  let nowStr = `${now.format('MM')}/${moment(plan.dateUpdated).format('DD')}/${now.format('YYYY')}`;
+  let nowStrFormat = 'MM/DD/YYYY';
 
   plan.dateTerminated =
-    moment(`${now.format('MM')}/${moment(plan.dateUpdated).format('DD')}/${now.format('YYYY')}`)
+    moment(nowStr, nowStrFormat)
     .add({days: remaining}) // end their subscription 1mo from their last payment
-    .add({months: Math.ceil(plan.extraMonths)})// plus any extra time (carry-over, gifted subscription, etc) they have. FIXME: moment can't add months in fractions...
+    .add({days: Math.ceil(30 * plan.extraMonths)}) // plus any extra time (carry-over, gifted subscription, etc) they have.
     .toDate();
   plan.extraMonths = 0; // clear extra time. If they subscribe again, it'll be recalculated from p.dateTerminated
 
-  data.user.save(cb);
+  data.user.save();
```

Comment:
we have to handle async ops like this somehow even if we're not using callbacks anymore, either return data.user.save(); ( a promise ) or transform this whole function in an async one